### PR TITLE
Add the crypto storage abstraction spec

### LINKED ISSUE #1820: Create a crypto storage API specification
Create a API specification:

Based on https://developer.hashicorp.com/vault/api-docs/secret/kv/kv-v1
* list secrets (might not be needed)
GET /secrets/keys/
* store / update secret
POST /secrets/keys/:id
* read secret
GET /secrets/keys/:id
* delete secret
DELETE /secrets/keys/:id

### PR DESCRIPTION
**Pull request by stevenvegt**
This PR adds the Crypto Storage Abstraction API Specification.

Fixes #1820 And is part of the #1810 project.

For easy review, I would advise to import the raw file into this online editor:
https://editor-next.swagger.io/

### CODE REVIEW

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: docs/_static/crypto/nuts-storage-api-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,199 @@
+openapi: "3.0.0"
+info:
+  title: Nuts Crypto Storage Abstraction API
+  description: |
+    API specification for a generic secret store HTTP server. 
+    It can be use to implement a secure store or a proxy to a storage solution such as 
+    Hashicorp Vault or Azure key store. Usually the Nuts node will be the client.
+    
+    The endpoints mounted at /secret act as a key-value store which accepts keys as path.
```
[the remark this answers is not in the document]
Comment (stevenvegt) [reply docs/_static/crypto/nuts-storage-api-v1.yaml:9]: What do you mean?

File: docs/_static/crypto/nuts-storage-api-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+              schema:
+                $ref: "#/components/schemas/ServiceStatus"
+        503:
+          description: |
+            The service is not healthy
+          content:
+            application/health+json:
+              schema:
+                $ref: "#/components/schemas/ServiceStatus"
+  /secrets:
```
[the remark this answers is not in the document]
Comment (stevenvegt) [reply docs/_static/crypto/nuts-storage-api-v1.yaml:37]: Ah, no they do not. It's just `/secrets`. I think you misunderstand the line in the description:
>     The endpoints mounted at /secret act as a key-value store which accepts keys as path.
So, I'm going to fix that :)

File: docs/_static/crypto/nuts-storage-api-v1.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (stevenvegt) [reply docs/_static/crypto/nuts-storage-api-v1.yaml]: Fixed, I removed that line.

File: docs/_static/crypto/nuts-storage-api-v1.yaml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+          description: The request failed. Check the error response for more details.
+          content:
+            application/problem+json:  # https://tools.ietf.org/html/rfc7807#section-6.1
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+  /secrets/{key}:
+    parameters:
+      - name: key
+        in: path
+        description: "Key under which the value should be stored. Note: The storage engine might encode these keys."
```
[the remark this answers is not in the document]
Comment (stevenvegt) [reply docs/_static/crypto/nuts-storage-api-v1.yaml]: I added more information, better like this?

File: docs/_static/crypto/nuts-storage-api-v1.yaml [hunk 4]
```diff
@@ -0,0 +1,199 @@
+openapi: "3.0.0"
```
[the remark this answers is not in the document]
Comment (stevenvegt) [reply docs/_static/crypto/nuts-storage-api-v1.yaml:1]: Currently this is a contract between the Nuts-node and an implementations, not much different from the current API specifications in sibling folders. I agree that the path is very obscure. A separate repository for all our api specifications would probably be more elegant, or event a less obscure path that the "docs" directory.

File: docs/_static/crypto/nuts-storage-api-v1.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (reinkrul) [reply docs/_static/crypto/nuts-storage-api-v1.yaml]: Not really, it still doesn't tell you what is allowed. If everything is allowed, fine too, just asking. If there are any restrictions, we should mention them (and if not, then it's OK).

**Review by reinkrul — APPROVED** [review]

**Pull request merged**